v8_test.go

```diff
@@ -1541,3 +1542,23 @@ func TestPanicHandling(t *testing.T) {
 	_ = NewIsolate()
 	_ = *f
 }
+
+func TestNewIsolateWithConstraints(t *testing.T) {
+	// Creates a v8 runtime where the memory is limited to 10MB and memory is
+	// allocated with a small script until v8 runs out of memory.
+	t.Parallel()
+	isolate := NewIsolateWithConstraints(ResourceConstraints{MaxOldSpaceSize: 10})
+	isolate.SetOOMErrorHandler(func(location string, isHeapOOM bool) {
+		// Success
+		os.Exit(0)
```

Comment:
This is not ok to do in a test since it'll interrupt other tests. If any other tests are failing, you might not see that because they are run in parallel but this one exits the entire test process before anything else fails.

Reply:
Glad you mentioned this. I assumed the same but didn't see the other tests stopping but didn't think of the implications of parallel test execution!

I am wondering if there's another way. Simply, without `os.Exit(0)` the test will fail because of the v8 runtime sending a non-0 exit code.

There's also a `SetFatalErrorHandler` we could set for v8, maye that would allow to override the behavior for the test scenario.

Reply:
If you need to test that the process crashes, then you can have this test run a separate test process:

https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go

Reply:
(though I'd prefer that we end up with a cleaner situation than crashing the process -- if you end up with a crash-the-process solution, please be VERY VERY clear in the docs about the side-effects!)